OpenRA.Mods.D2/Widgets/Logic/Ingame/D2IngameActorInfoDisplayLogic.cs

```diff
@@ -0,0 +1,277 @@
+			}
+
+			line2a = widget.GetOrNull<LabelWidget>("LINE2A");
+			if (line2a != null)
+			{
+				line2a.TextColor = textColor;
+				line2a.Font = "MediumBold";
+			}
+
+			line2b = widget.GetOrNull<LabelWidget>("LINE2B");
+			if (line2b != null)
+			{
+				line2b.Align = TextAlign.Right;
+				line2b.TextColor = textColor;
+				line2b.Font = "MediumBold";
+			}
+		}
+		public override void Tick()
+		{
+			base.Tick();
+
+			UpdateStateIfNecessary();
+		}
+
+		void HideExtraInfo()
+		{
+			if (title == null || separator == null || line1a == null || line1b == null || line2a == null || line2b == null)
+				return;
+
+			title.Visible = false;
+			separator.Visible = false;
+			line1a.Visible = false;
+			line1b.Visible = false;
+			line2a.Visible = false;
+			line2b.Visible = false;
+		}
+
+		void UpdateSpiceInfo(Actor actor)
+		{
+			if (title == null || separator == null || line1a == null || line1b == null || line2a == null || line2b == null)
+				return;
+
+			var stores = actor.TraitsImplementing<StoresResources>();
+			if (stores.Any())
+			{
+				var store = stores.FirstOrDefault();
+				if (store != null)
+				{
+					title.Text = "SPICE";
+					title.Visible = true;
+
+					separator.Visible = true;
+
+					line1a.Text = "HOLDS:";
+					line1a.Visible = true;
+					line1b.GetText = () => store.Stored.ToString();
+					line1b.Visible = true;
+
+					line2a.Text = "MAX:";
+					line2a.Visible = true;
+					line2b.GetText = () => (store as IStoreResources).Capacity.ToString();
+					line2b.Visible = true;
+				}
+			}
+		}
+
+		void UpdatePowerInfo(Actor actor)
+		{
+			if (title == null || separator == null || line1a == null || line1b == null || line2a == null || line2b == null)
+				return;
+
+			var powers = actor.TraitsImplementing<Power>();
+			if (powers.Any())
+			{
+				var power = powers.FirstOrDefault(t => !t.IsTraitDisabled);
+				if (power != null)
+				{
+					if (power.Info.Amount > 0)
+					{
+						title.Text = "POWER INFO";
+						title.Visible = true;
+
+						separator.Visible = true;
+
+						line1a.Text = "NEEDED:";
+						line1a.Visible = true;
+						line1b.GetText = () => (power.PlayerPower.PowerProvided != 0 ? power.PlayerPower.PowerDrained * power.GetEnabledPower() / power.PlayerPower.PowerProvided : 0).ToString();
+						line1b.Visible = true;
+
+						line2a.Text = "OUTPUT:";
+						line2a.Visible = true;
+						line2b.GetText = () => power.GetEnabledPower().ToString();
+						line2b.Visible = true;
+					}
+				}
+			}
+		}
+
+
+		void UpdateStateIfNecessary()
+		{
+			if (selectionHash == world.Selection.Hash)
+				return;
+
+			selectedActors = world.Selection.Actors
+				.Where(a => a.IsInWorld && !a.IsDead)
+				.ToArray();
+
+			if (selectedActors.Length == 1)
+			{
+				var actor = selectedActors[0];
+				if (label != null)
+				{
+					var tooltip = actor.TraitsImplementing<Tooltip>();
+					if (tooltip.Any())
+						label.Text = tooltip.FirstOrDefault(t => !t.IsTraitDisabled).Info.Name;
+					else
+					{
```

Comment:
Fwiw, you don't have braces around the other single-line statements, so you can remove them here.

Reply:
if I remember correctly, in OpenRA code style, braces should be set if it will be another closing brace, right after this one:

to not allow something like this:
```
						else
							label.Text = actor.Info.Name;
	}
```

but use 
```
						else
							label.Text = actor.Info.Name;
 						}
					}
				}
			}
		}                      
 	}
```

correct me, if I’m wrong

Reply:
Hm, I don't see that at https://github.com/OpenRA/OpenRA/wiki/Coding-Standard.
The only exception I know is when you have an inner block with more lines.
E.g.
```
if (condA)
    while (condB)
    {
    ...
    ...
    }
```
should be
```
if (condA)
{
    while (condB)
    {
    ...
    ...
    }
}
```

Reply:
ok. fixed